src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,160 @@
-
-(defn save-discover!
-  [{:keys [new-discover]} _]
-  (discoveries/save new-discover))
-
-(defn reload-tags!
-  [db _]
-  (assoc db :tags (discoveries/get-all-tags)
-            :discoveries (->> (discoveries/get-all :desc)
-                              (map (fn [{:keys [message-id] :as discover}]
-                                     [message-id discover]))
-                              (into {}))))
-
-(register-handler :add-discover
-  (u/handlers->
-    add-discover
-    save-discover!
-    reload-tags!))
-
-(register-handler :remove-old-discoveries!
-  (u/side-effect!
-    (fn [_ _]
-      (discoveries/delete :created-at :asc 1000 200))))
+(handlers/register-handler-fx
+  :broadcast-status
+  [(re-frame/inject-cofx :random-id)]
+  (fn [{{:keys [current-public-key web3]
+         :accounts/keys [accounts current-account-id]
+         :contacts/keys [contacts]} :db
+        random-id :random-id}
+       [_ status]]
+    (when-let [hashtags (seq (handlers/get-hashtags status))]
+      (let [{:keys [name photo-path]} (get accounts current-account-id)
+            message    {:message-id random-id
+                        :from       current-public-key
+                        :payload    {:message-id random-id
+                                     :status     status
+                                     :hashtags   (vec hashtags)
+                                     :profile    {:name          name
+                                                  :profile-image photo-path}}}]
+
+        {::broadcast-status {:web3       web3
+                             :message    message
+                             :identities (handlers/identities contacts)}
+         :dispatch          [:status-received message]}))))
+
+(handlers/register-handler-fx
+  :init-discoveries
+  (fn [_ _]
+    {:data-store.discover/get-all [#(re-frame/dispatch [:data-store/db-update {:discoveries %}])
+                                   maximum-number-of-discoveries]
+     :dispatch                    [:request-discoveries]}))
+
+(handlers/register-handler-fx
+  :request-discoveries
+  [(re-frame/inject-cofx :random-id)]
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db
+        random-id :random-id} _]
+    ;; this event calls itself at regular intervals
+    ;; TODO (yenda): this was previously using setInterval explicitly, with
+    ;; dispatch-later it is using it implicitly. setInterval is
+    ;; problematic for such long period of time and will cause a warning
+    ;; for Android in latest versions of react-nativexb
+    {::request-discoveries {:current-public-key current-public-key
+                            :web3               web3
+                            :identities         (handlers/identities contacts)
+                            :message-id         random-id}
+     :dispatch-later       [{:ms       (* request-discoveries-interval-s 1000)
+                             :dispatch [:request-discoveries]}]}))
```

Comment:
Instead of hardcoding the event name here, you can simply take the first item from the dispatch vector, it's less fragile that way (if you ever change `:request-discoveries` to something else, you don't have to change it here as well)

Reply:
good point